### PR TITLE
Fix faulty API endpoint

### PR DESCRIPTION
**Pull request by felixheidecke**
add more allowed characters in username generator

### CODE REVIEW

**Comment by SergioBertolinSG** [comment]
It is failing in my environment:

```javascript
jquery.js:8630 PUT http://HIDDEN_IP/html/owncloud/index.php/apps/files/ 405 (Method Not Allowed)
send @ jquery.js:8630
ajax @ jquery.js:8166
addGuest @ app.js?v=0001110…:101
boundFn @ vue.js?v=0001110…:130
invoker @ vue.js?v=0001110…:1641
VM2491:1 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (app.js?v=0001110…:105)
    at j (jquery.js:3099)
    at Object.fireWith [as rejectWith] (jquery.js:3211)
    at x (jquery.js:8266)
    at XMLHttpRequest.<anonymous> (jquery.js:8605)
```

**Comment by SergioBertolinSG** [comment]
This PR fixes https://github.com/owncloud/guests/issues/47

**Comment by SergioBertolinSG** [comment]
The problem is that the frontend is requesting ```apps/files``` instead of ```apps/guests/users```

File: js/app.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ $(document).ready(function () {
 				var self = this;
 				xhrObject = {
 					type: 'PUT',
-					url: '/index.php' + OC.linkTo('guests', 'users'),
+					url: OC.getRootPath() + '/index.php' + OC.linkTo('guests', 'users'),
```
Comment (PVince81) [inline js/app.js]: `OC.generateUrl('/apps/guests/users')` might help fix this

**Comment by PVince81** [comment]
@SergioBertolinSG please verify and merge if this works

**Comment by SergioBertolinSG** [comment]
Why change the way the username is formed?
 here it has change to dots '.' instead of underscores '_'.

**Comment by SergioBertolinSG** [comment]
The fix works fine.

Just the change in the username formation worrying me.  @IljaN if you agree with that change, we can merge.

**Comment by IljaN** [comment]
@SergioBertolinSG Not sure about the username really. Is LDAP or maybe some other system using the dot character as some kind of seperator?

@PVince81 Is it a potential problem to use a dot in the username?

If we ever reach the gold platting phase with this app, then @felixheidecke needs to write a real cool name generator ;)

**Comment by IljaN** [comment]
LDAP Special chars: http://www-12.lotus.com/ldd/doc/domino_notes/7.0/help_admin_upgrade7.nsf/b3266a3c17f9bb7085256b870069c0a9/b741831654e21bb4852570bb0050038f

**Pull request merged**